scripts/makebumpver.j2

```diff
@@ -0,0 +1,467 @@
+
+        self.name = self.args.name
+        self.version = self.args.version
+        self.release = self.args.release
+        self.new_release = self.args.new_release or self.release
+        self.bugreport = self.args.bugreporting_email
+        self.ignore = self.args.ignored_commits
+        self.skip = self.args.skip_all_checks
+        self.dry_run = self.args.dry_run
+
+        self.git_branch = None
+
+        # RHEL release number or None (also fills in self.git_branch)
+        self.rhel = self._is_RHEL()
+
+    def _git_config(self, field):
+        proc = run_program(['git', 'config', field])
+        return proc[0].strip('\n')
+
+    def _increment_version(self, bump_major_version=False, add_version_number=False):
+        fields = self.version.split('.')
+        if add_version_number:  # add another version number to the end
+            fields.append(str(int(DEFAULT_ADDED_VERSION_NUMBER)))
+        elif bump_major_version:  # increment major version
+            fields[0] = str(int(fields[0]) + VERSION_NUMBER_INCREMENT)  # bump major version
+            fields[1] = str(int(DEFAULT_MINOR_VERSION_NUMBER))  # reset minor version to 1
+        else:  # minor version bump
+            fields[-1] = str(int(fields[-1]) + VERSION_NUMBER_INCREMENT)
+        new = ".".join(fields)
+        return new
+
+    def _is_RHEL(self):
+        proc = run_program(['git', 'branch'])
+        lines = [x for x in proc[0].strip('\n').split('\n') if x.startswith('*')]
+
+        if lines == [] or len(lines) > 1:
+            return False
+
+        fields = lines[0].split(' ')
+
+        if len(fields) == 2:
+            self.git_branch = fields[1]
+        else:
+            return False
+
+        if self.git_branch.startswith('rhel'):
+            branch_pattern = r"^rhel(\d+)-(.*)"
+            m = re.match(branch_pattern, self.git_branch)
+            if m:
+                return m.group(1)
+            rhel8_branch_pattern = r"^rhel-(\d+)(.*)"
+            m = re.match(rhel8_branch_pattern, self.git_branch)
+            if m:
+                return m.group(1)
+        return False
+
+    def _get_commit_detail(self, commit, field):
+        proc = run_program(['git', 'log', '-1', "--pretty=format:%s" % field, commit])
+        ret = proc[0].strip('\n').split('\n')
+
+        if len(ret) == 1 and ret[0].find('@') != -1:
+            ret = [ret[0].split('@')[0]]
+        elif len(ret) == 1:
+            ret = [ret[0]]
+        else:
+            ret = [x for x in ret if x != '']
+
+        return ret
+
+    def _rpm_log(self, fixedIn):
+        git_range = "%s-%s-%s.." % (self.name, self.version, self.release)
+        proc = run_program(['git', 'log', '--no-merges', '--pretty=oneline', git_range])
+        lines = proc[0].strip('\n').split('\n')
+
+        for commit in self.ignore:
+            lines = [x for x in lines if not x.startswith(commit)]
+
+        rpm_log = []
+        bad_bump = False
+        bad = False
```

Comment:
Maybe we can get rid of one of the `bad` and `bad_bump` variables now.

Reply:
Good point.